mqtt/mqtt-edgehub/src/translation.rs

```diff
@@ -0,0 +1,377 @@
+/***********************************************************************************************************************
+ * This module translates between the old sdk iothub topics which lack a client id
+ * into newly designed topics that include a client id.
+ *
+ * IotHub currently relies on the connection information to identify a device.
+ * This does not work well in he broker message since edgehub core doesn't hold the connection to the device/module.
+ *
+ * This translation allows edgehub core to subscribe to the new topics that include client id to identify requests
+ ***********************************************************************************************************************
+ */
+
+use lazy_static::lazy_static;
+use mqtt3::proto;
+use regex::Regex;
+use tracing::debug;
+
+lazy_static! {
+    pub static ref TRANSLATOR: Translator = Translator::new();
```

Comment:
nit: maybe you don't even need an exposed instance of the Translator? why not make `incoming_subscribe, etc, just a module funcs and call them from Connection?

Reply:
I agree, Translator doesn't have any state, just functions. Now macros generate too many "unnecessary" internal structs, that just hold Regex. Could we try to simplify this a bit? Like, have a list of statically created Regexes for each required case and function? Probably you could make nested modules for groups in and out directions for the one topic? something like

```rust
pub mod publish {
  lazy_static! {
    REGEX_INCOMING = ,
    REGEX_OUTGOING = ,
}
  fn translate_incoming(...) -> Publish {
     if topic.starts_with("$iothub/") || topic.starts_with("devices/") {
        super::translate_incoming(REGEX_INCOMING, publish)
    }
  }

   fn translate_outgoing(...) -> Publish {
     if topic.starts_with("$edgehub/") {
        super::translate_outgoing(REGEX_INCOMING, publish)
    }
  }
}
  
```